lib/data_hygiene/delete_unneeded_topics.rb

```diff
@@ -38,43 +48,58 @@ def subscriber_lists
 
     def with_zero_subscribers
       @with_zero_subscribers ||= subscriber_lists
-        .select { |c, s| c&.zero? }
+        .select { |c, s| c == '0' }
+        .map(&:last)
+    end
+
+    def with_missing_topic
+      @with_missing_topic ||= subscriber_lists
+        .select { |c, s| c == :topic_not_found }
         .map(&:last)
     end
 
     def with_gov_delivery_error
       @with_gov_delivery_error ||= subscriber_lists
-        .select { |c, s| c.nil? }
+        .select { |c, s| c == :unknown_error }
         .map(&:last)
     end
 
   private
 
     def subscriber_count(gov_delivery_id)
       topic = client.fetch_topic(gov_delivery_id)
-      topic && topic['subscribers_count']
+      topic.subscribers_count
```

Comment:
It depends how much we want to trust GovDelivery to only return a number in this field - `'abc'.to_i` also returns `0` which might mean that we end up deleting some topics which returned some funky data instead of actually having 0 subscribers. I've added a comment about it in a separate commit.